torch/csrc/jit/ir.h

```diff
@@ -659,12 +673,20 @@ friend struct Node;
   }
 };
 
-inline Node::Node(Graph * graph_, NodeKind kind_) :
+inline Node::Node(Graph * graph_, NodeKind kind_, std::string pretty_name_) :
   kind_(kind_),
   graph_(graph_),
+  pretty_name_(pretty_name_),
   unique_(graph_->next_unique_++),
   stage_(graph_->new_node_stage_),
   type_(getInitialType(kind_)) {
+  if (pretty_name_ != "") {
+    if (graph_->unique_pretty_names.find(pretty_name_) != graph_->unique_pretty_names.end()) {
+      // TODO: how do I raise an error?
```

Comment:
You can use JIT_ASSERT(0) (or, more preferably, `JIT_ASSERT(graph_->unique_pretty_names.find(pretty_name_) != graph_->unique_pretty_names.end())`) to signify that there has been a fatal invariant broken, indicating developer error. Although, I'm not sure we should apply that here, because pretty names come from Python world, so they can trigger this condition themselves. In that case, raising a `std::runtime_error` is more appropriate.